File: 6-kyu/ip-validation/ip-validation.js
function isValidIP(str) {
  const segments = str.split('.');
  if (segments.length !== 4) return false;
  return segments.every(seg => {
    // Check if it's numeric, no leading zeros, within range
    return /^[0-9]+$/.test(seg) && 
           !(seg.length > 1 && seg.startsWith('0')) &&
           +seg >= 0 && +seg <= 255;
  });
}

File: 6-kyu/ip-validation/ip-validation.py
def is_valid_IP(ip):
    parts = ip.split('.')
    if len(parts) != 4:
        return False
    for part in parts:
        if not part.isdigit():
            return False
        if len(part) > 1 and part[0] == '0':
            return False
        num = int(part)
        if num < 0 or num > 255:
            return False
    return True